src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -764,5 +764,77 @@ void GcsActorManager::LoadInitialData(const EmptyCallback &done) {
   RAY_CHECK_OK(gcs_table_storage_->ActorTable().GetAll(callback));
 }
 
+void GcsActorManager::OnJobFinished(const JobID &job_id) {
+  RAY_CHECK_OK(gcs_table_storage_->ActorTable().DeleteByJobId(job_id, nullptr));
+  auto on_done =
+      [this](const std::unordered_map<ActorID, ActorCheckpointIdData> &result) {
+        if (!result.empty()) {
+          std::vector<ActorCheckpointID> checkpoint_ids;
+          std::vector<ActorID> actor_ids;
+          for (auto &item : result) {
+            actor_ids.push_back(item.first);
+            for (auto &id : item.second.checkpoint_ids()) {
+              checkpoint_ids.push_back(ActorCheckpointID::FromBinary(id));
+            }
+          }
+          RAY_CHECK_OK(gcs_table_storage_->ActorCheckpointTable().BatchDelete(
+              checkpoint_ids, nullptr));
+          ClearActorStates(actor_ids);
+        }
+      };
+  // Get checkpoint id first from checkpoint id table and delete all checkpoints related
+  // to this job
+  RAY_CHECK_OK(gcs_table_storage_->ActorCheckpointIdTable().GetByJobId(job_id, on_done));
+  RAY_CHECK_OK(
+      gcs_table_storage_->ActorCheckpointIdTable().DeleteByJobId(job_id, nullptr));
+}
+
+void GcsActorManager::ClearActorStates(std::vector<ActorID> actor_ids) {
+  for (auto &actor_id : actor_ids) {
```

Comment:
This logic looks like it duplicates the `DestroyActor` logic. Please use that method instead.

Also, I don't think that we actually need to call this method when the job finishes. If the driver exits, all actors related to the driver will eventually exit via fate-sharing. Once the actor's owner exits, the GCS actor manager will call `DestroyActor` anyway. Perhaps we should modify `DestroyActor` to also delete the Redis entry (which I notice you're not doing here, any reason for that?), but that seems low-priority right now, and we should wait for #8679 to be merged before we do that.

Reply:
At first I tried to reuse `DestroyActor` but it seems like will do some checks while what we need is to clear in-memory state.
After job finished we need to clear in-memory state and the entries(actor/checkpoint/checkpoint id table) in Redis of its actors. These two steps are now done in `OnJobFinished`.
As you said if all in-memory state will be cleared after driver exists, maybe we just delete Redis entry in `OnJobFinished` method.

Reply:
Yes, I think that makes sense (just deleting the Redis entry in OnJobFinished).